### DISCUSSION TITLE
Using ReadImage twice on one IM object (PerlMagick)

### DISCUSSION THREAD

**Discussion by ranknfyle** [ImageMagick/ImageMagick6#356] [General]
### ImageMagick version
6.8.9-8 Q16  Perl 5.16.3

### Operating system, version and so on
Win 10

### Description
This is about PerlMagick, I hope it's OK on this forum. When I "ReadImage" twice into the same IM object in Perl ($img), does the image in the $imgpath2 file overwrite the current contents of the IM object ($img)? Or does it create a new "page"/"layer" within $img? If it's the latter, how can I just overwrite the contents of $img with what's in $imgpath2, without new pages/layers? $imgpath1 & $imgpath2 have the same size.

```
my $img=Image::Magick->new();
$img->ReadImage($imgpath1);
$img->ReadImage($imgpath2);
```

**Comment by dlemstra** [comment]
It is an overwrite of the image information, it's not adding another page/layer
